### PR TITLE
Fix minor activity timer not queuing polls

### PR DESCRIPTION
**Pull request by Volco**
the ``ts-ignores`` are because when i added ``class: Poll`` it says some properties are missing and the other one is "type MinorActivity has no construct signatures". If you recommend handling the errors some way else lmk.

personally I wish it was possible to import poll.ts to room-minor-activity without it crashing on boot but if it was then the ``throw new Error`` line would have never been necessary

### CODE REVIEW

File: server/room-minor-activity.ts [hunk 1]
```diff
@@ -95,7 +96,10 @@ export abstract class MinorActivity {
 			});
 
 			if (!MinorActivityClass) {
-				if (pollData.activityid === 'poll') throw new Error('No minorActivity class provided');
+				// hacky fix to make poll queues work with timers
+				const Poll = pollData.class;
+				// @ts-ignore
+				if (pollData.activityid === 'poll') room.setMinorActivity(new Poll(room, pollData));
```
Comment (AnnikaCodes) [inline server/room-minor-activity.ts:102]: The reason you can't import `Poll` here is that `poll.ts` expects the `Rooms` global to be set up. A less hacky way to solve this is to utilize `require`.
```suggestion
				if (pollData.activityid === 'poll') {
					const {Poll} = require('./chat-plugins/poll');
					room.setMinorActivity(new Poll(room, pollData));
				}
```

Comment (AnnikaCodes) [reply server/room-minor-activity.ts:102]: As far as I know, `Rooms` will always exist by the time we're ending a MinorActivity, so this seems safe to me.

Comment (Volco) [reply server/room-minor-activity.ts:102]: let me test it real fast just to make sure. for some reason i though require weren't allowed in TS files. probably my memory just being bad per usual

**Review by AnnikaCodes — APPROVED** [review]
Looks good to me.

File: server/room-minor-activity.ts [hunk 2]
```diff
@@ -95,7 +95,10 @@ export abstract class MinorActivity {
 			});
 
 			if (!MinorActivityClass) {
-				if (pollData.activityid === 'poll') throw new Error('No minorActivity class provided');
```
Comment (Zarel) [inline server/room-minor-activity.ts:98]: I think I want to know more about why `MinorActivityClass` doesn't exist here. It definitely looks like we need some sort of global registry of minor activity classes (probably also games) to make this easier to use.

**Comment by Zarel** [comment]
Not a fix for the core problem, but it sure fixes the symptoms, so thanks!

**Pull request merged**